<launch>

<arg name="name" default="whycon"/>
<arg name="targets" default="1"/>
<arg name="outer_diameter" default=".55"/>
<arg name="inner_diameter" default=".20"/>
<arg name="input_queue_size" default="100"/>

<node name="whycon" type="whycon" pkg="whycon" output="screen">
  <param name="targets" value="$(arg targets)"/>
  <param name="outer_diameter" value="$(arg outer_diameter)"/>
  <param name="inner_diameter" value="$(arg inner_diameter)"/> 
  <param name="input_queue_size" value="$(arg input_queue_size)"/> 
  <param name="name" value="$(arg name)"/>

  <remap to="/visionSensor/camera_info" from="/camera/camera_info" />
  <remap to="/visionSensor/image_rect" from="/camera/image_rect_color" />
    
</node>

<!-- Uncomment the following when you want to run task 1.1 -->
<!-- <node name="task_1.1" type="task_1.1.py" pkg="pollinator_bee" output="screen"/> -->

<!-- Uncomment the following when you want to run task 1.2 -->
<!-- <node name="task_1.2" type="task_1.2.py" pkg="pollinator_bee" output="screen"/> -->

<!-- PID tune node -->
<!-- <node name="pid_tune_gui" type="pid_tune_gui.py" pkg="pid_tune" output="screen"/> -->

<!-- PlotJuggler node -->
<!-- <node name="plotJuggler" type="PlotJuggler" pkg="plotjuggler" output="screen"/> -->

<node name="image_view" pkg="image_view" type="image_view" respawn="false" output="screen">
  <remap from="image" to="/whycon/image_out"/>
  <param name="autosize" value="true" />
</node>

</launch>